tests/aws/services/s3/test_s3_list_operations.py

```diff
@@ -46,6 +48,87 @@ def assert_timestamp_is_iso8061_s3_format(timestamp: str):
     assert parsed_ts.microsecond == 0
 
 
+class TestS3ListBuckets:
+    @markers.aws.validated
+    def test_list_buckets_by_prefix_with_case_sensitivity(
+        self, s3_create_bucket, aws_client, snapshot
+    ):
+        snapshot.add_transformer(snapshot.transform.s3_api())
+
+        bucket_name = f"test.bucket.{short_uid()}"
```

Comment:
small nit: dots `.` are not well supported in S3, because it creates URL that have subdomains, and do not have SSL certs covering them. Might be better to use `-` instead.